pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -54,8 +54,8 @@ const (
 )
 
 var (
-	emptyStruct   struct{}
-	snapstoreHash = make(map[string]interface{})
+	emptyStruct          struct{}
+	snapstorePrevModTime time.Time
```

Comment:
While it's easier to use use a global variable for this, it is [generally not recommended](https://www.baeldung.com/cs/global-variables) unless you have a very strong case for it, and I would recommend adding this as part of `Snapshotter` struct, so that it is limited to the snapshotter object itself. This way, `snapstorePrevModTime` can be accessed by any methods called on the `Snapshotter` object.

Reply:
Thanks for the suggestion @shreyas-s-rao. I was following the previous design of having a global variable which is stores information on whether the credential files have changed. I'm willing to add `snapstorePrevModTime` to `Snapshotter`, and would like to get feedback from @ishan16696 and @seshachalam-yv regarding this suggested change.

Reply:
no strong opinion form my side

Reply:
Going ahead with removing the global variable `snapstorePrevModTime` and adding a field to the `Snapshotter` `struct`. 

Also taking the liberty to rename `snapstorePrevModTime` to `prevSecretModTime` since the variable this is compared with is `newSecretModTime` (which holds the latest modification time of the access credential files) and would thus improve readability.